modules/katello_content_view_version.py

```diff
@@ -0,0 +1,270 @@
+author: "Sean O'Keeffe (@sean797)"
+requirements:
+    - "nailgun"
+    - "python >= 2.6"
+notes:
+    - You cannot use this to remove a Contnet View Version from a Lifecycle environment, you should promote another version first.
+    - For idempotency you must specify either C(version) or C(current_lifecycle_environment).
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        default: true
+        type: bool
+    content_view:
+        description:
+            - Name of the content view
+        required: true
+    organization:
+        description:
+            - Organization that the content view is in
+        required: true
+    state:
+       description:
+            - Content View Version state
+       default: present
+       choices:
+           - absent
+           - present
+    version:
+        description:
+           - The content view version number (i.e. 1.0)
+    lifecycle_environments:
+        description:
+            - The lifecycle environments the Content View Version should be in.
+        default: Library
+    force_promote:
+        description:
+            - Force content view promotion and bypass lifecycle environment restriction
+        default: false
+        type: bool
+        aliases:
+          - force
+    force_yum_metadata_regeneration:
+        description:
+            - Force metadata regeneration when performing Publish and Promote tasks
+    synchronous:
+        description:
+            - Wait for the Publish or Promote task to complete if True. Immediately return if False.
+        default: true
+    current_lifecycle_environment:
+        description:
+            - The lifecycle environment that is already associated with the content view version
+            - Helpful for promoting a content view version
+'''
+
+EXAMPLES = '''
+- name: "Ensure content view version 2.0 is in Test & Pre Prod"
+  katello_content_view_version:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    content_view: "CV 1"
+    organization: "Default Organization"
+    version: 2.0
+    lifecycle_environments:
+      - Test
+      - Pre Prod
+
+- name: "Ensure content view version in Test is also in Pre Prod"
+  katello_content_view_version:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    content_view: "CV 1"
+    organization: "Default Organization"
+    current_lifecycle_environment: Test
+    lifecycle_environments:
+      - Pre Prod
+
+- name: "Publish a content view, not idempotent"
+  katello_content_view_publish:
```

Comment:
Is this still deprecated?

I like the concept of exposing CVVs as first-class objects in f-a-m a lot, but usually I think of the CV as a "cvv generator" and interact with the CV's API to create CVVs.